frontend/packages/ceph-storage-plugin/src/components/dashboards/ocs-system-dashboard.tsx

```diff
@@ -0,0 +1,201 @@
+
+const isPagePresent = (pages: Page[], page: Page): boolean =>
+  pages.some((p) => page.href === p.href);
+
+const sortPages = (a: Page, b: Page): number => {
+  if (a.href === 'block-file') return -1;
+  if (b.href === 'block-file') return 1;
+  return 0;
+};
+
+type CommonDashboardRendererProps = {
+  leftCards: React.ComponentType[];
+  rightCards: React.ComponentType[];
+  mainCards: React.ComponentType[];
+};
+
+const CommonDashboardRenderer: React.FC<CommonDashboardRendererProps> = ({
+  leftCards,
+  rightCards,
+  mainCards,
+}) => {
+  const mainGridCards: GridDashboardCard[] = mainCards.map(convertToCard);
+  const leftGridCards: GridDashboardCard[] = leftCards.map(convertToCard);
+  const rightGridCards: GridDashboardCard[] = rightCards.map(convertToCard);
+
+  return (
+    <Dashboard>
+      <DashboardGrid
+        mainCards={mainGridCards}
+        leftCards={leftGridCards}
+        rightCards={rightGridCards}
+      />
+    </Dashboard>
+  );
+};
+
+const PersistentInternalDashboard: React.FC = () => {
+  const mainCards: React.ComponentType[] = [
+    StatusCard,
+    RawCapacityCard,
+    BreakdownCard,
+    UtilizationCard,
+  ];
+  const leftCards: React.ComponentType[] = [DetailsCard, InventoryCard, storageEfficiencyCard];
+  const rightCards: React.ComponentType[] = [ActivityCard];
+
+  return (
+    <CommonDashboardRenderer leftCards={leftCards} mainCards={mainCards} rightCards={rightCards} />
+  );
+};
+
+const PersistentExternalDashboard: React.FC = () => {
+  const mainCards: React.ComponentType[] = [ExtStatusCard, ExtBreakdownCard, ExtUtilizationCard];
+  const leftCards: React.ComponentType[] = [DetailsCard, InventoryCard];
+  const rightCards: React.ComponentType[] = [ActivityCard];
+
+  return (
+    <CommonDashboardRenderer leftCards={leftCards} mainCards={mainCards} rightCards={rightCards} />
+  );
+};
+
+const ObjectServiceDashboard: React.FC = () => {
+  const mainCards: React.ComponentType[] = [
+    ObjectStatusCard,
+    ObjectBreakdownCard,
+    DataConsumptionCard,
+  ];
+  const leftCards: React.ComponentType[] = [
+    ObjectDetailsCard,
+    StorageEfficiencyCard,
+    BucketsCard,
+    ResourceProvidersCard,
+  ];
+  const rightCards: React.ComponentType[] = [ObjectActivityCard];
+  return (
+    <CommonDashboardRenderer leftCards={leftCards} mainCards={mainCards} rightCards={rightCards} />
+  );
+};
+
+const OCSSystemDashboard: React.FC<DashboardsPageProps> = ({
+  match,
+  kindsInFlight,
+  k8sModels,
+  history,
+}) => {
+  const isIndependent = useFlag(OCS_INDEPENDENT_FLAG);
+  const isObjectServiceAvailable = useFlag(MCG_FLAG);
+  const isCephAvailable = useFlag(CEPH_FLAG);
+  const [pages, setPages] = React.useState<Page[]>([]);
+  const { t } = useTranslation();
+  const title = t('ceph-storage-plugin~OpenShift Container Storage Overview');
+
+  const location = useLocation();
+
+  const isOCS = location.pathname.includes('ocs-dashboards');
```

Comment:
i see `ocs-dashboards` being hardcoded in lots of places. 
we can extract it out in a common place and use it